src/components/stepforms/ConditionsEditor/ConditionsGroup.vue

```diff
@@ -0,0 +1,470 @@
+<template>
+  <div
+    class="conditions-group"
+    :class="{
+      'conditions-group--with-switch': hasMultipleRows,
+    }"
+  >
+    <div v-if="hasMultipleRows" class="conditions-group__switch">
+      <div class="condition-group__switch-link" />
+      <div class="conditions-group__switch-buttons">
+        <div
+          class="conditions-group__switch-button conditions-group__switch-button--and"
+          :class="{ 'conditions-group__switch-button--active': operator === 'and' }"
+          @click="switchOperator('and')"
+        >
+          and
+        </div>
+        <div
+          class="conditions-group__switch-button conditions-group__switch-button--or"
+          :class="{ 'conditions-group__switch-button--active': operator === 'or' }"
+          @click="switchOperator('or')"
+        >
+          or
+        </div>
+      </div>
+    </div>
+    <div v-for="(condition, rowIndex) in conditions" :key="'row' + rowIndex" class="condition-row">
+      <div
+        v-if="hasMultipleRows"
+        class="condition-row__link"
+        :class="{
+          'condition-row__link--last':
+            !(groups && groups.length) && rowIndex === conditions.length - 1,
```

Comment:
This is a matter of taste but I find the expression a bit too hard to read at first sight :)

Reply:
Totally right, I refactored this into methods that explains the logic better.
I also added a few tests to ensure this logic won't break.
See fe0c1ed